### ISSUE TITLE
Language server not working in seperate source directory. (Using neovim)

### ISSUE THREAD

**Issue by yoghurt-x86** [elm-tooling/elm-language-server#395]
I have a elm.json with two source-directories that looks like:
```
{
    "type": "application",
    "source-directories": [
        "src",
	"../lib"
    ],
    "elm-version": "0.19.1",
    "dependencies": {
        "direct": {
            "elm/browser": "1.0.2",
            "elm/core": "1.0.5",
            "elm/html": "1.0.0",
            "elm/http": "2.0.0",
            "elm/json": "1.1.3"
        },
        "indirect": {
            "elm/bytes": "1.0.8",
            "elm/file": "1.0.5",
            "elm/time": "1.0.0",
            "elm/url": "1.0.0",
            "elm/virtual-dom": "1.0.2"
        }
    },
    "test-dependencies": {
        "direct": {},
        "indirect": {}
    }
}
```
Here is a picture of the directory structure. 
![image](https://user-images.githubusercontent.com/25183723/101753916-74fa4000-3ad3-11eb-96cd-ef6d3966a99d.png)

I am using neovim 0.5.0 with the build-in LSP client and elmls config from neovim/nvim-lspconfig 
It works as expected in normal projects, i can see warnings and jump to definitions and such, but if i open a file in a source directory like ../lib there is no lsp functionality

If i open Main.elm there is no issues. I can even jump to the definitions in ../lib/CustomLib.elm
With Main.elm open in the current buffer and i run:
`:lua print(vim.inspect(vim.lsp.buf_get_clients()))`
I get a bunch of output. 

If i jump to ../lib/CustomLib.elm and run the same command i get no output.
So it looks like the LSP client isn't attached?

If i instead move the directory ../lib inside the elm project (next to  src) then it will work as expected.
But in my case i want multiple elm projects to share the same lib.

Is it possible to get the LSP client to attach in another source directory outside the elm project?

**Comment by Razzeee** [comment]
I don't think we support this at the moment. It might work okayish, if you use vscode workspaces, but that's kind of a workaround.

**Comment by yoghurt-x86** [comment]
Alright, I've also tried to work around it using a symbolic link like:
![image](https://user-images.githubusercontent.com/25183723/101905448-c5e36480-3bb7-11eb-98d5-e33c81850c83.png)
But that doesn't work either it seems. Does symbolic links cause trouble?

**Comment by Razzeee** [comment]
Haven't tried that, might depend on your OS and how you create them

**Comment by yoghurt-x86** [comment]
I think i have found a solution that works. 
I have simply had to add a elm.json to the lib folder like it's a "package". Here is the folder structure:
![image](https://user-images.githubusercontent.com/25183723/101982037-7e211380-3c71-11eb-99aa-28bcd4107f15.png)

The elm.json for ./lib
```
{
    "type": "package",
	"name": "custom/custom",
	"summary": "For testing",
	"license": "MIT",
	"version": "1.0.0",
    "source-directories": [
        "src"
    ],
    "elm-version": "0.19.0 <= v < 0.20.0",
	"exposed-modules": {
        "Whatever": [
            "CustomLib"
		]
	},
    "dependencies": {
		"elm/core": "1.0.0 <= v < 2.0.0"
	},
    "test-dependencies": {}
}
```

the elm.json for the main project:
```
{
    "type": "application",
    "source-directories": [
        "src",
	"../lib/src"
    ],
    "elm-version": "0.19.1",
    "dependencies": {
        "direct": {
            "elm/browser": "1.0.2",
            "elm/core": "1.0.5",
            "elm/html": "1.0.0"
        },
        "indirect": {
            "elm/json": "1.1.3",
            "elm/time": "1.0.0",
            "elm/url": "1.0.0",
            "elm/virtual-dom": "1.0.2"
        }
    },
    "test-dependencies": {
        "direct": {},
        "indirect": {}
    }
}
```

What's weird is that the original project that i started with works fine in vscode, but this was neeed to make it work in neovim.

**Comment by Razzeee** [comment]
I think the problem is, that your opening the folder where your `elm.json` is and we only do our completions on files in there. For e.g. if you go to one of your dependencies in elm/core I think you loose functionality too.

Basically means this is kind of a duplicate of https://github.com/elm-tooling/elm-language-server/issues/368

**Comment by yoghurt-x86** [comment]
I can kind of cheat it to work by making a dummy `elm.json` that i place in the root folder next to the proj folder. 
Now the lsp client is attached when i open up files in ./lib.
![image](https://user-images.githubusercontent.com/25183723/101987342-cea96880-3c93-11eb-8a67-70a4faa4006b.png)

I'm not sure if #368 is related. I can jump to definitions in the Browser module for example, and the lsp client will be attached there, and i can further navigate to other code definitions from there. 
This files are writable though on my system.

**Comment by Razzeee** [comment]
Yes, they are writable, I just choose to differentiate between packages you can change and packages you shouldn't change (thirdparty ones from your cache) and named that bool `writeable`

**Comment by Razzeee** [comment]
> I'm not sure if #368 is related. I can jump to definitions in the Browser module for example, and the lsp client will be attached there, and i can further navigate to other code definitions from there.

It turned out to actually be a problem in the vscode client, so your probably fine in neovim :)
